app/code/Magento/InventoryCatalog/Model/GetProductIdsBySourceItemIds.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Model;
+
+/**
+ * Provide corresponding product ids for given source item ids.
+ */
+class GetProductIdsBySourceItemIds
```

Comment:
Why do we need some `proxy` class without any logic?

Reply:
@naydav, removed.